src/Sylius/Bundle/AttributeBundle/Resources/public/js/sylius-attribute.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+(function ($) {
+    'use strict';
+
+    function calculateNextIndex() {
+        var nextIndex = 0;
+        var attributeItems = $('#attributes > div .collection-item');
+
+        if (attributeItems.length == 0) {
```

Comment:
I am not sure if JS will throw here some error in case of property assign but maybe you should consider yoda condition style :D ?